jobs/payment-jobs/tasks/statement_due_task.py

```diff
@@ -117,11 +117,16 @@ def _notify_for_monthly(cls):
                         # The locking email is sufficient for overdue, no seperate email required.
                         additional_emails = current_app.config.get('EFT_OVERDUE_NOTIFY_EMAILS')
                         AuthEvent.publish_lock_account_event(payment_account, additional_emails)
-                        cls.add_to_non_sufficient_funds(payment_account)
                         statement.overdue_notification_date = datetime.now(tz=timezone.utc)
+                        # Saving here because the code below can exception, we don't want to send the lock email twice.
                         statement.save()
+                        cls.add_to_non_sufficient_funds(payment_account)
```

Comment:
I've seen `cfs_account.cfs_account` doesn't exist  we don't want to spam the lock account if that occurs, so I've moved this up.. there's a bit of a risk